executor/src/witgen/affine_expression.rs

```diff
@@ -206,54 +282,53 @@ where
         &self,
         known_constraints: &impl RangeConstraintSet<K, T>,
     ) -> EvalResult<T, K> {
-        let parts = self
+        let unconstrained: Vec<K> = self
+            .nonzero_coefficients()
+            .filter_map(|(i, _coeff)| known_constraints.range_constraint(i).is_none().then_some(i))
+            .collect();
+
+        if !unconstrained.is_empty() {
+            return Ok(EvalValue::incomplete(IncompleteCause::BitUnconstrained(
+                unconstrained,
+            )));
+        }
+
+        let masks = self
             .nonzero_coefficients()
             .map(|(i, coeff)| {
                 (
                     i,
                     *coeff,
-                    known_constraints
+                    *known_constraints
                         .range_constraint(i)
                         .unwrap()
```

Comment:
Because if unconstrained is non-empty, the function returns in line 292 - or did I see it wrong?
Maybe I'll try to re-write it to create the masks and return at the same time.

Reply:
Oh. I knew I was missing something... It is fine by me the way it is.

Reply:
I changed the code to compute the masks and the unconstrained items at the same time.

Reply:
It might be nice though to see if we can fail early, because computing the masks is expensive.

Reply:
oh, found an easy way.